aspects/schema.go

```diff
@@ -232,17 +233,17 @@ type mapSchema struct {
 func (v *mapSchema) Validate(raw []byte) error {
 	var mapValue map[string]json.RawMessage
 	if err := json.Unmarshal(raw, &mapValue); err != nil {
-		return err
+		return &validationError{baseErr: err}
 	}
 
 	if mapValue == nil {
-		return fmt.Errorf(`cannot accept null value for "map" type`)
+		return &validationError{baseErr: fmt.Errorf(`cannot accept null value for "map" type`)}
```

Comment:
is it worth having helpers like:

```
validationErrorWith(err)

validationErrorf(format, ...)
```

instead of always building the struct by hand?

Reply:
There used to be a `newValidationError(err)` but since we only set the baseErr I didn't think it was necessary. I can add some helpers to make the calls tidier